deployment/cdk/opensearch-service-migration/lib/stack-composer.ts

```diff
@@ -301,6 +335,68 @@ export class StackComposer {
             this.stacks.push(mskUtilityStack)
         }
 
+        let migrationAnalyticsStack;
+        let analyticsDomainStack;
+        if (migrationAnalyticsServiceEnabled && networkStack) {
+            const analyticsDomainName = "migration-analytics-domain"
+            const openAccessPolicy = new PolicyStatement({
+                effect: Effect.ALLOW,
+                principals: [new AnyPrincipal()],
+                actions: ["es:*"],
+                resources: [`arn:aws:es:${region}:${account}:domain/${analyticsDomainName}/*`]
+            })
+            analyticsDomainStack = new OpenSearchDomainStack(scope, `analyticsDomainStack`,
+            {
+                stackName: `OSMigrations-${stage}-${region}-AnalyticsDomain`,
+                description: "This stack prepares the Migration Analytics OS Domain",
+                domainAccessSecurityGroupParameter: "analyticsDomainSGId",
+                endpointParameterName: "analyticsDomainEndpoint",
+                version: this.getEngineVersion(analyticsDomainEngineVersion ?? engineVersion),  // If no analytics version is specified, use the same as the target cluster
```

Comment:
nit: I'd like to use the opensearch latest by default, but understand there isn't a built in way to get this